src/Compilers/CSharp/Portable/BoundTree/BoundNodes.xml

```diff
@@ -1599,17 +1668,28 @@
          free. The necessity of this member is a consequence of a design issue documented in
          https://github.com/dotnet/roslyn/issues/13960 . When that is fixed this field can be
          removed. -->
-    <Field Name="VariableAccess" Type="BoundExpression" Null="disallow"/>
+    <Field Name="VariableAccess" Type="BoundExpression" Null="allow"/>
     <Field Name="DeclaredType" Type="BoundTypeExpression" Null="allow"/>
     <Field Name="IsVar" Type="bool"/>
   </Node>
 
-  <Node Name="BoundConstantPattern" Base="BoundPattern">
-    <Field Name="Value" Type="BoundExpression"/>
-    <Field Name="ConstantValue" Type="ConstantValue" Null="allow"/>
-  </Node>
-
-  <Node Name="BoundWildcardPattern" Base="BoundPattern">
+  <Node Name="BoundRecursivePattern" Base="BoundPattern">
+    <Field Name="DeclaredType" Type="BoundTypeExpression" Null="allow"/>
+    <Field Name="InputType" Type="TypeSymbol"/>
+    <Field Name="DeconstructMethodOpt" Type="MethodSymbol" Null="allow"/>
+    <Field Name="Deconstruction" Type="ImmutableArray&lt;BoundPattern&gt;" Null="allow"/>
+    <Field Name="PropertiesOpt" Type="ImmutableArray&lt;(Symbol symbol, BoundPattern pattern)&gt;" Null="allow"/>
```

Comment:
>PropertiesOpt [](start = 17, length = 13)

The name is misleading, this is not just about properties, consider renaming to MembersOpt

Reply:
This represents a property subpattern, per the current spec, so I'm keeping the name. I agree we should consider renaming the language construct, and the bound node for it at the same time.

---
In reply to: [156536108](https://github.com/dotnet/roslyn/pull/23209#discussion_r156536108) [](ancestors = 156536108)